File: Martensa_Backend/Backend/microservices-parent/authService/src/main/java/org/demo/authservice/service/AuthService.java
package org.demo.authservice.service;

import jakarta.validation.Valid;
import org.demo.authservice.dto.AuthResponse;
import org.demo.authservice.dto.LoginRequest;
import org.demo.authservice.dto.RegisterRequest;
import org.springframework.http.ResponseEntity;

public interface AuthService {
    AuthResponse register(RegisterRequest registerRequest);

    AuthResponse login(@Valid LoginRequest request);

    boolean validateToken(String token);

    AuthResponse getToken(String email);
}

File: Martensa_Frontend/modules/auth/services/authService.ts
// modules/auth/services/authService.ts
import apiClientNoAuth from "@/services/apiClientNoAuth";
import apiClient from "../../../services/apiClient";
import { AuthResponse, LoginRequest, RegisterRequest } from "../types/auth";

export const login = async (data: LoginRequest): Promise<AuthResponse> => {
  const response = await apiClientNoAuth.post<AuthResponse>(
    "/auth/login",
    data
  );
  return response.data;
};

export const register = async (
  data: RegisterRequest
): Promise<AuthResponse> => {
  const response = await apiClientNoAuth.post<AuthResponse>(
    "/auth/register",
    data
  );
  return response.data;
};

export const validateToken = async (token: string) => {
  return await apiClient.post("/auth/validate", null, {
    headers: {
      Authorization: `Bearer ${token}`,
    },
  });
};
